### PR TITLE
Checkfor  body params necessary to create a video. return message 

### PR DESCRIPTION
**Pull request by jtwray**
# Description
### first pass solution
- checks request.body params against a list of preset required values.
- returns the first one missing.
- cleans up and replaces if-else "callback hell" in the router's response logic 

# Motivation 
Improves readability in any router that has required req.body elements

## Type of change

- [ x ] Bug fix (non-breaking change which fixes an issue)

## Change Status

- [ x ] Complete, tested, ready to review and merge
- [ x ] first pass solution

# How Has This Been Tested?

- [ x ] Test A  tested locally in postman on my machine
- [ x ] Test B tested on zoom call with @panamajake86 

# Checklist

- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] My code has been reviewed by at least one peer
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

**Review by panamajake86 — APPROVED** [review]
Very nice. This code is pretty elegant. Love the middleware solution. Glad to not have to see those nested if/else statements anymore. Approved

File: middleware/checkfor.js [hunk 1]
```diff
@@ -1,48 +1,10 @@
...
-                res.status(400).json({ message: "Missing video_title!" });
-              } else {
-                if (!user_id) {
-                  res.status(400).json({ message: "Missing user_id!" });
\ No newline at end of file
+      ? next()
+      : res
+          .status(400)
+          .json({
+            errorMessage: `The ${prop} is required to create a video. Please include a ${prop} and try again.`,
```
Comment (hisnameisjimmy) [inline middleware/checkfor.js:8]: This is cool as hell, I really like this

Comment (hisnameisjimmy) [reply middleware/checkfor.js:8]: It's also gonna cause hella merge conflicts when my was s3checker comes down the pipe

File: videos/videos_router.js [hunk 2]
```diff
@@ -33,105 +36,70 @@ router.get("/:id", async (req, res) => {
   }
 });
 
-router.get("/user/:userId", async (req, res) => {
```
Comment (jimmique-parsons) [inline videos/videos_router.js:36]: It looks like the path to find videos by the user was erased by accident possibly? 
```
router.get("/user/:userId", async (req, res) => {
  const { userId } = req.params;
  try {
    const userVideos = await Videos.findByUser(userId);
    res.status(200).json(userVideos);
  } catch (err) {
    res.status(500).json({ message: "Try again later.", err });
  }
}); 
```
I believe we are currently using this to post the user's videos to their profile page. I'll send up a commit with it added back into the file.

**Review by jimmique-parsons — APPROVED** [review]

File: videos/videos_router.js (hunk 2 above)
Comment (jtwray) [reply videos/videos_router.js:36]: Good Catch. Thanks.

Comment (jtwray) [reply videos/videos_router.js]: Turns out the getbyUserID route was on master. Its in this version for sure after the merge resolution.

**Pull request merged**